### PR TITLE
Fake GPU query success for Xdefiant

### PR DESCRIPTION
**Pull request by jp7677**
Not tested ...

### CODE REVIEW

File: src/util/util_env.cpp [hunk 1]
```diff
@@ -135,6 +135,10 @@ namespace dxvk::env {
         return getExecutableName() == std::string("aces.exe");
     }
 
+    bool isXDefiant() {
+        return getExecutableName() == std::string("upc.exe");
```
Comment (Saancreed) [inline src/util/util_env.cpp]: `upc.exe` is Ubisoft Connect itself, game executable is called `XDefiant.exe` as far as I can tell.

Comment (Saancreed) [reply src/util/util_env.cpp]: Whoops, already fixed, ignore me :sweat_smile:

File: src/util/util_env.cpp [hunk 2]
```diff
@@ -154,12 +154,23 @@ namespace dxvk::env {
             || getExecutableName() == std::string("tlou-i-l.exe");
     }
 
+
+    bool isXDefiant() {
+        return getExecutableName() == std::string("XDefiant.exe")
+            || getExecutableName() == std::string("XDefiant_BE.exe");
```
Comment (Saancreed) [inline src/util/util_env.cpp:160]: I think the latter is not needed but I suppose we can keep it just to be safe.

File: src/util/util_env.cpp [hunk 3]
```diff
@@ -154,12 +154,23 @@ namespace dxvk::env {
...
+    }
+
     bool needsSucceededGpuQuery() {
         if (isTheLastOfUsPartOne()) {
             log::write("Faking GPU query success due to detecting tlou-i.exe/tlou-i-l.exe");
             return true;
         }
 
+        if (isXDefiant()) {
+            log::write("Faking GPU query success due to detecting upc.exe (XDefiant)");
```
Comment (Saancreed) [inline src/util/util_env.cpp:170]: Still has old/wrong binary name in the log message.

**Review by Saancreed — APPROVED** [review]
:ok_hand:, assuming this is indeed enough to make it work.

**Comment by xpander69** [comment]
Works great here now. Thanks!
```
grep "N[v-V]API" ~/Xdefiant-newNVAPI
NvAPI_QueryInterface (0xad298d3f): Unknown function ID
DXVK-NVAPI v0.7.0-10-g4477057 (XDefiant.exe)
NvAPI Device: NVIDIA GeForce RTX 3080 (555.42.2)
NvAPI Output: \\.\DISPLAY1
NvAPI Output: \\.\DISPLAY2
NvAPI_Initialize: OK
NvAPI_QueryInterface (0x33c7358c): Unknown function ID
NvAPI_QueryInterface (0x593e8644): Unknown function ID
NvAPI_SYS_GetDriverAndBranchVersion: OK
NvAPI_EnumPhysicalGPUs: OK
NvAPI_GPU_GetPCIIdentifiers: OK
NvAPI_GPU_GetFullName: OK
NvAPI_GPU_GetGpuCoreCount: OK
NvAPI_GPU_GetArchInfo: OK
NvAPI_D3D_SetLatencyMarker: No implementation
NvAPI_D3D_SetSleepMode: No implementation
NvAPI_EnumPhysicalGPUs: OK
NvAPI_GPU_GetPCIIdentifiers: OK
NvAPI_GPU_GetCurrentPstate: OK
NvAPI_GPU_GetAllClockFrequencies: OK
NvAPI_GPU_GetDynamicPstatesInfoEx: OK
NvAPI_GPU_GetThermalSettings: OK
```

**Review by Saancreed — APPROVED** [review]

**Pull request merged**